### ISSUE TITLE
Doc Shell overhaul

### ISSUE THREAD

**Issue by endigo9740** [skeletonlabs/skeleton#977]
Per: https://github.com/skeletonlabs/skeleton/issues/783

To kick off the new documentation updates, one of the first items we need to address is a cleanup and refactor of the Doc Shell component. This and the "Previewer" component will be critical to be in place before we can begin improving examples, editing the copy, etc.

## Todo

- [x] Break this up into a small set of components for easier maintenance
- [x] Improved layout, (flatter?) navigation, and better UX as relevant
- [x] Implement structure changes, such as a dedicated examples/usage sections
- [x] Provide a better home for the table of contents - it's just sort of duct tape on right now
- [x] Implement the new ad position in the sidebar above table of contents
- [x] Support the ability to deep link to section headers
- [x] Create a reusable class to replace `space-y-4` assigned to section Section element

## References:

- Radix UI Docs - https://www.radix-ui.com/docs/primitives/components/accordion
- VitePress - https://vitepress.vuejs.org/
- Tailwind - https://tailwindcss.com/docs/installation
- Docusaurus - https://docusaurus.io/
- Chakra UI - https://chakra-ui.com/docs/components/button/props

**Comment by endigo9740** [comment]
So great progress today. Here's a preview of before vs after comparing the new design structure. Note that nothing shown here is set in stone yet.

![Screenshot 2023-02-21 at 5 27 05 PM](https://user-images.githubusercontent.com/1509726/220482714-c19bfa22-8653-4806-bcaa-45a808e76522.png)

![Screenshot 2023-02-21 at 5 27 11 PM](https://user-images.githubusercontent.com/1509726/220482720-5a4d1d98-de1b-4218-b181-85689fe14879.png)

Notable changes:

- Makes much better use of the available space, reduces all the empty whitespace in the heading
- The import statement is now much more prominent
- Features an explicit "examples" area near the top 
- Tabs are more localized (should support sticky-top soon)

More to come tomorrow!

**Comment by niktek** [comment]
Looking good - only nitpick I can come up with is maybe changing the Github label to be `Component Source` with the same GH logo as the icon. Makes it balance with the Doc Source button and also slightly more full justifies the line

**Comment by endigo9740** [comment]
More progress today:

![Screenshot 2023-02-22 at 5 32 15 PM](https://user-images.githubusercontent.com/1509726/220787337-3d6707c5-c92a-4b14-9068-ac3ee39bc001.png)

- All heading features are now functional
- I've combined the import and type imports into a single code block
- Each tab "panel" (prop, classes, etc) is now functional
- Updated a few pages missing the new LayoutPage (sidebar + table of contents)
- I've implemented the staging version of the Carbon ads and added theme styles 
- Adjusted some feature examples to utilize the new format better

Overall this should be working MUCH closer to the final product. Tomorrow I'll finalize this and aim to merge into dev branch asap. Implemented the new Previewer component will be the next order of business.
